app/src/main/java/com/github/palFinderTeam/palfinder/ProfileViewModel.kt

```diff
@@ -94,4 +109,30 @@ class ProfileViewModel @Inject constructor(
         }
     }
 
+    /**
+     * We handle the block/unblock logic here to avoid handling coroutines in
+     * the adapter. Notice that we need to fetch again the user who wants to follow,
+     * that's because we need to have the more up to date info.
+     */
+    fun block(userId: String, otherId: String) {
```

Comment:
nit-peek: you could merge the block and unblock into a block toggle and thus reduce the amount of dupplicated code. but again just nit-peeking

Reply:
it improve coverage